Refresh.GameServer/Services/AipiService.cs

```diff
@@ -0,0 +1,161 @@
+using System.Diagnostics;
+using System.Net.Http.Json;
+using Bunkum.Core.Services;
+using JetBrains.Annotations;
+using NotEnoughLogs;
+using Refresh.GameServer.Configuration;
+using Refresh.GameServer.Importing;
+using Refresh.GameServer.Types.Assets;
+using Refresh.GameServer.Types.Data;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Processing;
+
+namespace Refresh.GameServer.Services;
+
+// Referenced from DO.
+public class AipiService : EndpointService
+{
+    private readonly HttpClient _client;
+    private readonly IntegrationConfig _config;
+    private readonly DiscordStaffService? _discord;
+
+    private readonly ImageImporter _importer;
+    
+    [UsedImplicitly]
+    public AipiService(Logger logger, IntegrationConfig config, ImportService import, DiscordStaffService discord) : base(logger)
+    {
+        this._discord = discord;
+        this._config = config;
+
+        this._client = new HttpClient
+        {
+            BaseAddress = new Uri(config.AipiBaseUrl),
+        };
+
+        this._importer = import.ImageImporter;
+    }
+
+    public override void Initialize()
+    {
+        if (!this._config.DiscordStaffWebhookEnabled)
+        {
+            this.Logger.LogWarning(RefreshContext.Aipi, 
+                "The Discord staff webhook is not enabled, but AIPI is. This is probably behavior you don't want.");
+        }
+        this.TestConnectivityAsync().Wait();
+    }
+
+    private async Task TestConnectivityAsync()
+    {
+        try
+        {
+            HttpResponseMessage response = await this._client.GetAsync("/");
+            string content = await response.Content.ReadAsStringAsync();
+
+            if (response.IsSuccessStatusCode && content == "AIPI scanning service")
+                this.Logger.LogInfo(RefreshContext.Aipi, "AIPI appears to be working correctly");
+            else
+                this.Logger.LogError(RefreshContext.Aipi,
+                    $"AIPI seems to be down. Status code: {response.StatusCode}, content: {content}");
+        }
+        catch (Exception e)
+        {
+            this.Logger.LogError(RefreshContext.Aipi, "AIPI connection failed: {0}", e.ToString());
+        }
+    }
+    
+    private async Task<TData> PostAsync<TData>(string endpoint, Stream data)
+    {
+        HttpResponseMessage response = await this._client.PostAsync(endpoint, new StreamContent(data));
+        AipiResponse<TData>? aipiResponse = await response.Content.ReadFromJsonAsync<AipiResponse<TData>>();
+        
+        if (aipiResponse == null) throw new Exception("No response was received from the server.");
+        if (!aipiResponse.Success) throw new Exception($"{response.StatusCode}: {aipiResponse.Reason}");
```

Comment:
There should be an `AipiException` class, or the exception text should elaborate that it comes from AIPI, it makes grepping logs much easier.

Reply:
Errors are more of a formality; you're unlikely to actually get one (as it's mostly going to be a server error which is logged in more detail anyways), so this is fine for now.